penguin/tools/core/support.py

```
@@ -961,12 +947,9 @@ def apply_diff_to_file(
 
         except Exception as e:
-            # Attempt to restore from backup
-            if backup:
+            if write_happened:
                 try:
-                    backup_path = target_path.with_suffix(target_path.suffix + ".bak")
-                    if backup_path.exists():
-                        shutil.copy2(backup_path, target_path)
+                    target_path.write_bytes(raw)
                 except Exception:
                     pass
```

Ruff (0.15.12)

[warning] Use `contextlib.suppress(Exception)` instead of `try`-`except`-`pass`

Replace `try`-`except`-`pass` with `with contextlib.suppress(Exception): ...`

(SIM105)